src/graph_visualization.cpp

```diff
@@ -0,0 +1,449 @@
+#include <SDL2/SDL.h>
+#include <iostream>
+#include <vector>
+
+using namespace std;
+
+#define SCREEN_WIDTH 640
+#define SCREEN_HEIGHT 480
+#define DISTANCE 40
+
+struct tree_node {
+    using node_id = int;
+    vector<node_id> succ = {};
+    vector<node_id> pred = {};
+    node_id id = 0; // no
+    int deg_p = 0;
+    int deg_m = 0;
+    int layer = 0;
+    int number = 0;
+    int x = 0;
+    int y = 0;
+};
+
+struct Net {
+    using node_id = tree_node::node_id;
+    
+    const vector<node_id> &get_sources();
+    const vector<node_id> &get_sinks();
+    
+    const tree_node *get_node(node_id id);
+};
```

Comment:
The definitions that are going to be reused have to be defined in a header file.